File: code_combat/4_Backwoods_Forest/186-Cursed_Wonderglade/cursed_wonder.py
while True:
    if hero.canCast("haste"):
        hero.cast("haste", hero)
    item = hero.findNearestItem()
    if item and item.type != "gem":
        if hero.isReady("jump"):
            hero.jumpTo(item.pos)
        else:
            hero.moveXY(item.pos.x, item.pos.y)

    enemy = hero.findNearestEnemy()
    if enemy and enemy.type != "burl":
        hero.attack(enemy)

File: projects/code_combat/4_Backwoods_Forest/186-Cursed_Wonderglade/cursedWonder.js
// Wonderglade has changed since our last visit.
// Ogres cursed it and we should defeat them.
// The burl still is collecting gems, so don't touch them.
// Also don't attack the burl.

while (true) {
    // Find the nearest item.
    // Collect it (if it exists) only if its type isn't "gem".
    var item = hero.findNearestItem();
    if (item && item.type != "gem") {
        hero.moveXY(item.pos.x, item.pos.y);
    }
    // Find the nearest enemy.
    // Attack it if it exists and its type isn't "burl".
    var enemy = hero.findNearestEnemy();
    if (enemy && enemy.type != "burl") {
        hero.attack(enemy);
    }
}
